#781 clustering otherCatalogNumbers

timrobertson100 opened the pull request:
**DO NOT MERGE THIS!**

This brings a new check that allows publishers to force a relationship when an `otherCatalogNumber` is provided in a format that matches an `institutionCode:collectionCode:catalogNumber` on the other side. See #781 for discussion on this.

Is there anything in here that alarms you? I don't think this will cause issue for ALA @djtfmartin 

I am running a final analysis before we commit this

djtfmartin commented:
thanks @timrobertson100  - i cant see any issues from our side.

MattBlissett reviewed (approved):
This looks fine to me.

timrobertson100 commented:
Thank you @djtfmartin @MattBlissett 

I have added a couple of NPE checks that were necessary when testing on all data, and then rebased and squashed to a single commit - i.e. I haven't substantially changed this - so will merge it.
